MonitoringSystem/Service/MonitoringHandler.py

```diff
@@ -0,0 +1,263 @@
+  
+  def __generatePlotFromFileId( self, fileId ):
+    """
+    It create the plots using the encode parameters
+    :param str fileId the encoded plot attributes
+    :return S_OK or S_ERROR returns the file name 
+    """
+    
+    result = extractRequestFromFileId( fileId )
+    if not result[ 'OK' ]:
+      return result
+    plotRequest = result[ 'Value' ]
+    gLogger.info( "Generating the plots.." )
+    result = self.export_generatePlot( plotRequest )
+    if not result[ 'OK' ]:
+      gLogger.error( "Error while generating the plots", result[ 'Message' ] )
+      return result
+    fileToReturn = 'plot'
+    if 'extraArgs' in plotRequest:
+      extraArgs = plotRequest[ 'extraArgs' ]
+      if 'thumbnail' in extraArgs and extraArgs[ 'thumbnail' ]:
+        fileToReturn = 'thumbnail'
+    gLogger.info( "Returning %s file: %s " % ( fileToReturn, result[ 'Value' ][ fileToReturn ] ) )
+    return S_OK( result[ 'Value' ][ fileToReturn ] )
+  
+  def __sendErrorAsImg( self, msgText, fileHelper ):
+    """
+    In case of an error message a whcite plot is created with the error message.
+    """
+    
+    retVal = generateErrorMessagePlot( msgText )
+    if not retVal:
+      retVal = fileHelper.sendData( retVal[ 'Message' ] )
+    else:
+      retVal = fileHelper.sendData( retVal[ 'Value' ] )
+      if not retVal[ 'OK' ]:
+        return retVal
+      fileHelper.sendEOF()
+    return S_OK()
+
+  def __checkPlotRequest( self, reportRequest ):
+    """
+    It check the plot attributes. We have to make sure that all attributes which are needed are provided.
+    
+    :param dict reportRequest contains the plot attributes.
+    
+    """
+    # If extraArgs is not there add it
+    if 'extraArgs' not in reportRequest:
+      reportRequest[ 'extraArgs' ] = {}
+    if not isinstance( reportRequest[ 'extraArgs' ], self.__reportRequestDict[ 'extraArgs' ] ):
+      return S_ERROR( "Extra args has to be of type %s" % self.__reportRequestDict[ 'extraArgs' ] )
+    reportRequestExtra = reportRequest[ 'extraArgs' ]
+    
+    # Check sliding plots
+    if 'lastSeconds' in reportRequestExtra:
+      try:
+        lastSeconds = long( reportRequestExtra[ 'lastSeconds' ] )
+      except ValueError:
+        gLogger.error( "lastSeconds key must be a number" )
+        return S_ERROR( "Value Error" )
+      #TODO: Maybe we can have last hour in the monitoring
+      if lastSeconds < 3600:
+        return S_ERROR( "lastSeconds must be more than 3600" )
+      now = Time.dateTime()
+      reportRequest[ 'endTime' ] = now
+      reportRequest[ 'startTime' ] = now - datetime.timedelta( seconds = lastSeconds )
+    else:
+      # if end date is not there, just set it to now
+      if not reportRequest.get( 'endTime', False ):
+        reportRequest[ 'endTime' ] = Time.dateTime()
+    # Check keys
+    for key in self.__reportRequestDict:
+      if not key in reportRequest:
+        return S_ERROR( 'Missing mandatory field %s in plot reques' % key )
```

Comment:
Yes, but you can see there are other checks and as soon as a mandatory field exists we exit from the loop